python/dgl/contrib/graph_store.py

```diff
@@ -0,0 +1,355 @@
+        ndata_infos = self.proxy.list_ndata()
+        for name, shape, dtype in ndata_infos:
+            self._init_ndata(name, shape, dtype)
+
+        edata_infos = self.proxy.list_edata()
+        for name, shape, dtype in edata_infos:
+            self._init_edata(name, shape, dtype)
+
+    def __del__(self):
+        if self.proxy is not None:
+            self.proxy.terminate()
+
+    def _init_ndata(self, ndata_name, shape, dtype):
+        assert self.number_of_nodes() == shape[0]
+        shape = utils.toindex(np.array(shape, dtype=np.int64))
+        data = _CAPI_DGLCreateSharedMem(_get_ndata_path(self._graph_name, ndata_name),
+                                        shape.todgltensor(), dtype, "zero", False)
+        dlpack = data.to_dlpack()
+        self.ndata[ndata_name] = F.zerocopy_from_dlpack(dlpack)
+
+    def init_ndata(self, ndata_name, shape, dtype):
+        """Create node embedding.
+
+        It first creates the node embedding in the server and maps it to the current process
+        with shared memory.
+
+        Parameters
+        ----------
+        ndata_name : string
+            The name of node embedding
+        shape : tuple
+            The shape of the node embedding
+        dtype : string
+            The data type of the node embedding.
+        """
+        self.proxy.init_ndata(ndata_name, shape, dtype)
+        self._init_ndata(ndata_name, shape, dtype)
+
+    def destroy(self):
+        """Destroy the graph store.
+
+        This notifies the server that this client has terminated.
+        """
+        if self.proxy is not None:
+            self.proxy.terminate()
+        self.proxy = None
+
+
+def create_graph_store_server(graph_data, graph_name, store_type, num_workers,
+                              multigraph=False, edge_dir='in', port=8000):
+    """Create the graph store server.
+
+    The server loads graph structure and node embeddings and edge embeddings
+    in shared memory.
+
+    After the server runs, the graph store clients can access the graph data
+    with the specified graph name.
+
+    Parameters
+    ----------
+    graph_data : graph data
+        Data to initialize graph. Same as networkx's semantics.
+    graph_name : string
+        Define the name of the graph.
+    store_type : string
+        The type of the graph store.
+    num_workers : int
+        The number of workers that will connect to the server.
+    multigraph : bool, optional
+        Whether the graph would be a multigraph (default: False)
+    edge_dir : string
+        the edge direction for the graph structure.
+    port : int
+        The port that the server listens to.
+
+    Returns
+    -------
+    SharedMemoryStoreServer
+        The graph store server
+    """
+    return SharedMemoryStoreServer(graph_data, edge_dir, graph_name, multigraph,
+                                   num_workers, port)
+
+def create_graph_store_client(graph_name, store_type, port=8000):
```

Comment:
`create_graph_from_store` since it is returning a `DGLGraph` and users could run stuff like `in_degrees()` on it.